tests/program/linspace_tests.py

```diff
@@ -63,6 +63,81 @@ def test_output(self):
         assert_vm_output_almost_equal(self, self.output, vm.history)
 
 
+class PrePostDepTest(TestCase):
+    def setUp(self):
+        hold = ConstantPT(10 ** 6, {'a': '-1. + idx * 0.01'})
+        hold_random = ConstantPT(10 ** 5, {'a': -.4})
+        # self.pulse_template = (hold_random@(hold_random@hold).with_repetition(10)@hold_random@hold)\
+        self.pulse_template = (hold_random @ hold.with_repetition(10)).with_iteration('idx', 200)
+
+        self.program = LinSpaceIter(
+            length=200,
+            body=(
+                LinSpaceHold(bases=(-.4,), factors=(None,), duration_base=TimeType(10**5), duration_factors=None),
+                LinSpaceRepeat(body=(
+                LinSpaceHold(bases=(-1.,),factors=((0.01,),),duration_base=TimeType(10**6),duration_factors=None),
+                ), count=10),
+                # LinSpaceHold(bases=(-.4),factors=None,duration_base=TimeType(10**6),duration_factors=None),
+                # LinSpaceHold(bases=(-1.,),factors=((0.01,),),duration_base=TimeType(10**6),duration_factors=None)
+            ),)
+        
+
+        key = DepKey.from_voltages((0.01,), DEFAULT_INCREMENT_RESOLUTION)
+
+        self.commands = [
+            Set(channel=0, value=-0.4, key=DepKey(factors=())),
+             Wait(duration=TimeType(100000, 1)),
+             #here is what currently happens:
+             Set(channel=0, value=-1.0, key=DepKey(factors=(10000000,))),
+             Wait(duration=TimeType(1000000, 1)),
+             LoopLabel(idx=0, count=9),
+             Wait(duration=TimeType(1000000, 1)),
+             LoopJmp(idx=0),
+             #however, i think this is what should happen (maybe also with an additional "Set" before,
+             #which might cause complications if omitted in other contexts like AWG amplitude:
+             #LoopLabel(idx=0, count=10),
+             #Set(channel=0, value=-1.0, key=DepKey(factors=(10000000,))),
```

Comment:
This set statement is not required. because it is the same in every iteration and can be lifted out of the loop. Why should it happen? This is exactly one intended case the post != pre condition is supposed to detect and optimize.

Reply:
the thing is that it is required to do this twice because the initial state changes. I brooded over this quite a while I remember :D But it might still be that I am wrong...

Reply:
I don‘t think you‘re wrong; just in the very pathological case of repeating a sequence of 1000 pulses, where one element is just „Set“ with a Set command, one may want to prefer to have this codeblock only once and does not want that perceived optimization of omitting the Set command but having to write the repeated sequence twice.
But I don‘t think this is very relevant.